### PR TITLE
feat(interactive): support string as edge property type

### LINKED ISSUE #3405: feat(interactive): Support string property on edge
Support using string type on edge property.

### PR DESCRIPTION
**Pull request by liulx20**
Support string type property on edge. 

Fix #3405

### CODE REVIEW

**Review by luoxiaojian — DISMISSED** [review]

File: flex/interactive/examples/modern_graph/modern_graph_string_edge.yaml [hunk 1]
```diff
@@ -0,0 +1,71 @@
+name: modern_graph # then must have a modern dir under ${data} directory
```
Comment (zhanglei1949) [inline flex/interactive/examples/modern_graph/modern_graph_string_edge.yaml:1]: I suggest moving this testing schema.yaml to `flex/test/rt_mutable_graph`, which is the place we put testing related stuffs.

File: flex/tests/rt_mutable_graph/string_edge_property_test.cc [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+    auto oe =
+        graph.get_outgoing_edges(src_label_, src_lid, dst_label_, edge_label_);
+    CHECK(oe != nullptr) << "Got nullptr oe\n";
+    CHECK(oe->get_data().type == PropertyType::kString)
+        << "Inconsistent type, Except: string, Got " << oe->get_data().type;
+    std::cout << oe->get_data().AsStringView() << "\n";
+    LOG(INFO) << "Finish test get edge\n";
+  }
+
+  void test_get_graphview() {
```
Comment (zhanglei1949) [inline flex/tests/rt_mutable_graph/string_edge_property_test.cc:48]: Same

**Review by luoxiaojian — DISMISSED** [review]

File: flex/storages/rt_mutable_graph/mutable_csr.h [hunk 3]
```diff
@@ -254,34 +490,33 @@ class MutableCsrBase {
 
 template <typename EDATA_T>
 class TypedMutableCsrConstEdgeIter : public MutableCsrConstEdgeIterBase {
-  using nbr_t = MutableNbr<EDATA_T>;
+  using nbr_t_ptr = typename MutableNbrSlice<EDATA_T>::nbr_t_ptr;
```
Comment (zhanglei1949) [inline flex/storages/rt_mutable_graph/mutable_csr.h:493]: I suggest indicating the const ptr in the type name, i.e. `using nbr_const_ptr_t`

File: flex/storages/rt_mutable_graph/mutable_csr.cc [hunk 4]
```diff
@@ -35,9 +35,6 @@ template class MutableCsr<uint32_t>;
 template class SingleMutableCsr<Date>;
 template class MutableCsr<Date>;
 
-template class SingleMutableCsr<std::string>;
```
Comment (zhanglei1949) [inline flex/storages/rt_mutable_graph/mutable_csr.cc:38]: `template class SingleMutableCsr<std::string>` or `template class SingleMutableCsr<std::string_view>` or both?

**Pull request merged**